tests/integration/test_dashboards.py

```diff
@@ -220,3 +220,35 @@ def test_dashboards_deploys_dashboard_with_widget_title_and_description(ws, make
     sdk_dashboard = dashboards.deploy_dashboard(lakeview_dashboard, dashboard_id=sdk_dashboard.dashboard_id)
 
     assert ws.lakeview.get(sdk_dashboard.dashboard_id)
+
+
+def test_dashboards_deploys_dashboard_from_query_with_cte(ws, make_dashboard, tmp_path):
+    sdk_dashboard = make_dashboard()
+
+    query_with_cte = """
+WITH data AS (
+  SELECT
+    Address,
+    City,
+    State,
+    `Zip Code`,
+    Country
+  FROM
+  VALUES
+    ('160 Spear St 15th Floor', 'San Francisco', 'CA', '94105', 'USA'),
+    ('756 W Peachtree St NW, Suite 03W114', 'Atlanta', 'GA', '30308', 'USA'),
+    ('500 108th Ave NE, Suite 1820', 'Bellevue', 'WA', '98004', 'USA'),
+    ('125 High St, Suite 220', 'Boston', 'MA', '02110', 'USA'),
+    ('2120 University Ave, Suite 722', 'Berkeley', 'CA', '94704', 'USA') AS tab(Address, City, State, `Zip Code`, Country)
+)
+-- --title 'Databricks Office Locations'
```

Comment:
Is this a feature or a bug? We use sqlglot to find the first comment above the select for which we create a visualization. Usually this is the first line in the query file, but with CTE's that is not the case. 

Options:
1. We always read the first line, check if it starts with `--` after `lstrip`, then we parse it. The downside is that this would not work with multiline comments `/* .... */`
2. We keep it like this and update the documentation that the comment must be above the query to be visualized